src/map.cpp

```diff
@@ -6038,6 +6038,44 @@ bool map::sees( const tripoint &F, const tripoint &T, const int range, int &bres
     return visible;
 }
 
+int map::obstacle_coverage( const tripoint &loc1, const tripoint &loc2 )
+{
+    if( move_cost( loc2 ) > 2 && !has_flag_ter( TFLAG_FLAT, loc2 ) ) { // Can't hide if you are standing on something.
+        return 0;
+    }
+    tripoint obstaclepos;
+    int dx = loc1.x - loc2.x;
+    int dy = loc1.y - loc2.y;
+    int ax = std::abs( dx ) * 2;
+    int ay = std::abs( dy ) * 2;
+    int offset = std::min( ax, ay ) - ( std::max( ax, ay ) / 2 );
+    bresenham( loc2, loc1, offset, 0, [&obstaclepos, &loc2]( const tripoint & new_point ) {
+        if( new_point.x == loc2.x &&
+            new_point.y == loc2.y ) { // Only adjacent tile between you and enemy is checked for cover.
+            return true;
+        }
+        obstaclepos = new_point;
+        return false;
+    } );
+    auto obstacle_f = furn( obstaclepos ).obj();
+    auto obstacle_t = ter( obstaclepos ).obj();
+    const optional_vpart_position vp = veh_at( obstaclepos );
+    vehicle *const veh = !vp ? nullptr : &vp->vehicle();
+    const int part = veh ? vp->part_index() : -1;
+    if( obstacle_f.id ) {
+        return obstacle_f.coverage;
+    }
+    if( veh != nullptr ) {
+        const vpart_position vp( const_cast<vehicle &>( *veh ), part );
+        if( vp.obstacle_at_part() ) {
```

Comment:
Same as above (rearrangement to prevent nesting and declaring variables far from their utilization).

Reply:
Thanks, rearranged this part.